packages/editor/src/lib/exports/StyleEmbedder.ts

```diff
@@ -53,9 +58,14 @@ export class StyleEmbedder {
 			? getDefaultStylesForTagName(element.tagName.toLowerCase())
 			: NO_STYLES
 
-		const parentStyles = shouldSkipInheritedParentStyles
-			? (this.styles.get(element.parentElement as Element)?.self ?? NO_STYLES)
-			: NO_STYLES
+		let parentStyles = NO_STYLES
+		if (shouldSkipInheritedParentStyles) {
+			let el = element.parentElement
+			while (el) {
+				parentStyles = Object.assign({}, this.styles.get(el)?.self, parentStyles)
+				el = el.parentElement
+			}
+		}
```

Comment:
maybe minor, maybe not, but as these are called for every element in the dom, for large trees i could see this (allocating a new object and iterating over everything that came before to assign to it) getting pretty expensive for large/deep trees. The ideal thing here I think would be to build up the parent styles object as we traverse down the tree embedding styles so we don't have to constantly traverse up again. That's obvs a big change though, so maybe for now we could replace the `Object.assign` with a for/in and normal assignment so we don't end up with so many intermediate copies of the style map?